contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -0,0 +1,211 @@
+  }
+
+  /**
+   * @notice withdraws all funds availbale in this contract to the address specified
+   * @param to address where to send the funds
+   * @dev only owner can call this
+   */
+  function withdrawFundsTo(
+    address payable to
+  ) 
+    external
+    onlyOwner() 
+  {
+    to.transfer(address(this).balance);
+  }
+
+  /**
+   * @notice sets gasAccessController
+   * @param gasAccessController new gasAccessController contract address
+   * @dev only owner can call this
+   */
+  function setGasAccessController(
+    address gasAccessController
+  )
+    external
+    onlyOwner
+  {
+    setGasAccessControllerInternal(gasAccessController);
+  }
+  
+  /**
+   * @notice sets Arbitrum gas configuration
+   * @param maxSubmissionCost maximum cost willing to pay on L2
+   * @param maximumGasPrice maximum gas price to pay on L2
+   * @param gasCostL2 value to send to L2 to cover gas fee
+   * @param refundableAddress address where gas excess on L2 will be sent
+   * @dev access control provided by s_gasConfigAccessController
+   */
+  function setGasConfiguration(
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  )
+    external
+  {
+    require(s_gasConfigAccessController.hasAccess(msg.sender, msg.data), "Only gas configuration admin can call");
+    setGasConfigurationInternal(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+  }
+
+  
+  /**
+   * @notice validate method updates the state of an L2 Flag in case of change on the Arbitrum Sequencer. A zero answer considers the service as offline
+   * In case the previous answer is the same as the current it does not trigger any tx on L2
+   * In other case, a retryable ticket is created on the Arbitrum L1 Inbox contract. The tx gas fee can be paid from this contract providing a value, or the same address on L2
+   * @param previousAnswer previous aggregator answer
+   * @param currentAnswer new aggregator answer
+   * @dev access control provided internally by SimpleWriteAccessController
+   */
+  function validate(
+    uint256 /* previousRoundId */,
+    int256 previousAnswer,
+    uint256 /* currentRoundId */,
+    int256 currentAnswer
+  ) 
+    external 
+    override
+    checkAccess() 
+    returns (bool) 
+  {
+    // Avoids resending to L2 the same tx on every call
+    if (previousAnswer == currentAnswer) {
+      return true;
+    }
+
+    s_arbitrumInbox.createRetryableTicket{value: s_gasConfig.gasCostL2}(
+      s_flagsAddress, 
+      0, // L2 call value
+      s_gasConfig.maxSubmissionCost, // Max submission cost of sending data length
+      s_gasConfig.refundableAddress, // excessFeeRefundAddress
+      s_gasConfig.refundableAddress, // callValueRefundAddress
+      s_L2GasLimit,
+      s_gasConfig.maximumGasPrice, 
+      currentAnswer == 0 ? s_raiseFlagData : s_lowerFlagData
+    );
+    return true;
+  }
+  
+  function setGasConfigurationInternal(
```

Comment:
Internal and private functions should have an underscore prefix as per style guide.